### PR TITLE
[E2E] Add e2e for FederatedResourceQuota sync

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Add e2e for FederatedResourceQuota sync.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/policy/federatedresourcequota_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+				err := karmadactl.RunJoin(os.Stdout, karmadaConfig, opts)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+
+			ginkgo.By(fmt.Sprintf("waiting federatedResourceQuota(%s/%s) present on cluster: %s", frqNamespace, frqName, clusterName), func() {
+				clusterClient, err := util.NewClusterClientSet(clusterName, e2e.ControlPlaneClient, nil)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+				gomega.Eventually(func(g gomega.Gomega) (bool, error) {
+					_, err := clusterClient.KubeClient.CoreV1().ResourceQuotas(frqNamespace).Get(context.TODO(), frqName, metav1.GetOptions{})
```
Comment (huone1) [inline test/e2e/policy/federatedresourcequota_test.go:128]: the data of  resource quota is also checked

Comment (XiShanYongYe-Chang) [reply test/e2e/policy/federatedresourcequota_test.go:128]: I plan to add a case to test status collection, it will cover this situation.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**